### PR TITLE
Make GH tags consistent with GR tags

### PR DESCRIPTION
**Pull request by nilsvu**
## Proposed changes

Template on DataType and default to Frame::Inertial.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

**Review by kidder — CHANGES_REQUESTED** [review]
SpatialDim and VolumeDim do not have to be equal.  SpatialDim is a property of tensor-like objects.  VolumeDim is a property of the objects in Domain.   I may want to evolve 3-dimensional tensors on a 1-dimensional Domain.

**Comment by wthrowe** [comment]
Also `VolumeDim` is used when working with boundary structures when `Dim` would be ambiguous with the boundary dimension.

**Comment by nilsvu** [comment]
Ok I just dropped that commit for now.

File: src/Evolution/Systems/GeneralizedHarmonic/Constraints.hpp [hunk 1]
```diff
@@ -34,16 +34,16 @@ namespace gh {
  * \f$C_{iab} = \partial_i\psi_{ab} - \Phi_{iab},\f$ which is
  * given by Eq. (26) of \cite Lindblom2005qh
  */
-template <typename DataType, size_t SpatialDim, typename Frame>
```
Comment (kidder) [inline src/Evolution/Systems/GeneralizedHarmonic/Constraints.hpp]: drop this change

Comment (kidder) [reply src/Evolution/Systems/GeneralizedHarmonic/Constraints.hpp]: you still have a bunch of SpatialDim -> Dim changes in this file

File: src/Evolution/Systems/GeneralizedHarmonic/Tags.hpp [hunk 2]
```diff
@@ -149,34 +147,34 @@ struct EvolvedFieldsFromCharacteristicFields : db::SimpleTag {
  * `ThreeIndexConstraintCompute`, `FourIndexConstraintCompute`, and
  * `ConstraintEnergyCompute` respectively
  */
-template <size_t SpatialDim, typename Frame>
```
Comment (kidder) [inline src/Evolution/Systems/GeneralizedHarmonic/Tags.hpp:152]: don't change SpatialDim to Dim (multiple times in this file)

File: src/Evolution/Systems/GeneralizedHarmonic/TagsDeclarations.hpp [hunk 3]
```diff
@@ -11,47 +11,47 @@ namespace gh {
...
+template <typename DataType, size_t Dim, typename Frame = Frame::Inertial>
 struct CharacteristicSpeeds;
-template <size_t Dim, typename Frame>
+template <typename DataType, size_t Dim, typename Frame = Frame::Inertial>
 struct CharacteristicFields;
-template <size_t Dim, typename Frame>
+template <typename DataType, size_t Dim, typename Frame = Frame::Inertial>
 struct EvolvedFieldsFromCharacteristicFields;
 
-template <size_t SpatialDim, typename Frame>
```
Comment (kidder) [inline src/Evolution/Systems/GeneralizedHarmonic/TagsDeclarations.hpp:44]: again...

**Review by kidder — CHANGES_REQUESTED** [review]

File: src/Evolution/Systems/GeneralizedHarmonic/Tags.hpp (hunk 2 above)
Comment (nilsvu) [reply src/Evolution/Systems/GeneralizedHarmonic/Tags.hpp:152]: Changed back, but note that the use of `Dim` and `SpatialDim` is very inconsistent, even in this file

**Review by kidder — DISMISSED** [review]

**Review by kidder — APPROVED** [review]

**Comment by nilsvu** [comment]
Unrelated timeout

**Pull request merged**